hbase-server/src/main/java/org/apache/hadoop/hbase/HDFSBlocksDistribution.java

```diff
@@ -280,6 +280,18 @@ public long getBlocksLocalWithSsdWeight(String host) {
   private long getBlocksLocalityWeightInternal(String host, Visitor visitor) {
     long localityIndex = 0;
     HostAndWeight hostAndWeight = this.hostAndWeights.get(host);
+    // Compatible with local mode, see HBASE-24569
+    if (hostAndWeight == null) {
+      String currentHost = "";
+      try {
+        currentHost = DNS.getDefaultHost("default", "default");
```

Comment:
Is this time consuming?

Reply:
Comment above: "In my local env, the code below cost about 1μs~200μs, avg<5μs, much lesser than 1ms."

```
      try {
        currentHost = DNS.getDefaultHost("default", "default");
      } catch (Exception e) {
        // Just ignore, it's ok, avoid too many log info
      }
```

Reply:
Can we store the currentHost at some place? Usually it will not change?

Reply:
```
  public static String getDefaultHost(@Nullable String strInterface, @Nullable String nameserver, boolean tryfallbackResolution) throws UnknownHostException {
    if (strInterface != null && !"default".equals(strInterface)) {
      if (nameserver != null && "default".equals(nameserver)) {
        nameserver = null;
      }

      String[] hosts = getHosts(strInterface, nameserver, tryfallbackResolution);
      return hosts[0];
    } else {
      return cachedHostname;
    }
```
It cached in DNS yet.